### DISCUSSION TITLE
Binded assertion idea

### DISCUSSION THREAD

**Discussion by satoved** [livewire/livewire#5027] [2. Show and tell]
Sometime I tend to do weird assertions to make sure something is rendered. E.g. asserting that I can see a button text, or a field label. But then the test fails if I change the text on the page, which is, of course, not right.

What I really want to know, is that the HTML contains needed connection to Livewire. So that when I do:

```
Livewire::test(Component::class)->set('foo', 'bar');
```

I can be sure that it's actually doable on the page itself:

```
Livewire::test(Component::class)->assertBound('foo')->set('foo', 'bar');
```

Here's a code I use in the TestCase:

```
        TestableLivewire::macro('assertBound', function (string $property) {
            PHPUnit::assertMatchesRegularExpression(
                <<<PATTERN
/(wire:model[.\dA-Za-z]*=["']{$property}["']|entangle\(["']{$property}["']\))/
PATTERN,
                $this->stripOutInitialData($this->lastRenderedDom),
                "Property $property is not bound: HTML code doesn't have related wire:model for it."
            );

            return $this;
        });
```

It supports both wire:model and @entangle.

I was thinking to do the same for actions:

```
Livewire::test(Component::class)->assertActionable('submit')->call('submit');
```

Will be happy to make a PR for these!

**Comment by joshhanley** [comment]
@satoved have a look at https://github.com/christophrumpel/missing-livewire-assertions as it has a `->assertPropertyWired('email')` assertion and same for methods.

Hope this helps!

**Reply by satoved** [reply]
Oh, thanks! Exactly what I needed

**Reply by joshhanley** [reply]
@satoved no worries! 🙂
